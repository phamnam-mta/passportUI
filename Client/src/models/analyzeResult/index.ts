import { V3_0_3_AnalyzeResult } from "./v3_0_3";

export type AnalyzeResponse = {
    createdDateTime: string;
    lastUpdatedDateTime: string;
    status: string;
    analyzeResult: V3_0_3_AnalyzeResult;
};

export type StudioAnalyzeResult = {
    apiVersion: string;
    modelId: string;
    stringIndexType: StringIndexType;
    content: string;
    pages: StudioDocumentPage[];
    blocks?: StudioBlock[];
    paragraphs?: StudioParagraph[];
    tables?: StudioDocumentTable[];
    keyValuePairs?: StudioDocumentKeyValuePair[];
    entities?: StudioDocumentEntity[];
    styles?: StudioDocumentStyle[];
    documents?: StudioDocumentResult[];
};

type StringIndexType = "textElements" | "unicodeCodePoint" | "utf16CodeUnit";

export type StudioDocumentPage = {
    pageNumber: number;
    angle: number;
    width: number;
    height: number;
    unit: "pixel" | "inch";
    words: StudioWord[];
    images?: StudioImage[];
    selectionMarks?: StudioSelectionMark[];
    lines?: StudioLine[];
    spans?: StudioSpan[];
    kind?: string;
};

export type StudioDocumentTable = {
    rowCount: number;
    columnCount: number;
    cells: StudioDocumentTableCell[];
    boundingRegions: BoundingRegion[];
    spans: StudioSpan[];
};

export type StudioDocumentKeyValuePair = {
    key: StudioKeyValuePairElement;
    value?: StudioKeyValuePairElement;
    confidence: Confidence;
};

export type StudioDocumentEntity = {
    category: string;
    subCategory?: string;
    content: string;
    boundingRegions: BoundingRegion[];
    confidence: Confidence;
    spans: StudioSpan[];
};

export type StudioDocumentStyle = {
    confidence?: Confidence;
    spans: StudioSpan[];
    isHandwritten?: boolean;
};

export type StudioDocumentResult = {
    docType: string;
    boundingRegions: BoundingRegion[];
    fields: { [fieldName: string]: StudioFieldValue };
    confidence: Confidence;
    spans: StudioSpan[];
};

export type BoundingRegion = {
    pageNumber: number;
    polygon: Polygon;
};

export type StudioWord = {
    content: string;
    polygon: Polygon;
    confidence?: Confidence;
    span?: StudioSpan;
};

export type StudioImage = {
    pageNumber: number;
    confidence?: Confidence;
    span: StudioSpan;
};

export type StudioLine = {
    content: string;
    polygon: Polygon;
    spans: StudioSpan[];
};

export type StudioBlock = {
    kind: string;
    BoundingRegion: BoundingRegion[];
    content: string;
    spans: StudioSpan[];
    confidence: Confidence;
};

export type StudioParagraph = {
    spans: StudioSpan[];
    boundingRegions?: BoundingRegion[];
    content: string;
    role?: string;
};

export type StudioSelectionMark = {
    state: SelectionMarkState;
    polygon: Polygon;
    confidence?: Confidence;
    span?: StudioSpan;
};

export type StudioDocumentTableCell = {
    kind?: DocumentTableCellKind;
    rowIndex: number;
    columnIndex: number;
    rowSpan?: number;
    columnSpan?: number;
    content: any;
    boundingRegions: BoundingRegion[];
    spans?: StudioSpan[];
};

type StudioKeyValuePairElement = {
    content: string;
    boundingRegions: BoundingRegion[];
    spans: StudioSpan[];
};

export type StudioFieldValue = {
    type: FieldValueType;
    valueString?: string;
    valueDate?: string;
    valueTime?: string;
    valuePhoneNumber?: string;
    valueNumber?: number;
    valueInteger?: number;
    valueSelectionMark?: SelectionMarkState;
    valueSignature?: DocumentSignatureType;
    valueCountryRegion?: string;
    valueCurrency?: DocumentCurrencyType;
    valueArray?: StudioFieldValue[];
    valueObject?: { [name: string]: StudioFieldValue };
    valueAddress?: { [name: string]: string };
    content?: string;
    boundingRegions?: BoundingRegion[];
    confidence?: Confidence;
    spans?: StudioSpan[];
};

export type StudioSpan = {
    offset: number;
    length: number;
};

export type Polygon = [number, number, number, number, number, number, number, number];

type DocumentCurrencyType = {
    currencySymbol?: string;
    currency?: string;
    amount?: number;
};

export type SelectionMarkState = "selected" | "unselected";

type Confidence = number;

type DocumentTableCellKind = "content" | "rowHeader" | "columnHeader" | "stub" | "description";

type FieldValueType =
    | "string"
    | "date"
    | "time"
    | "phoneNumber"
    | "number"
    | "integer"
    | "selectionMark"
    | "countryRegion"
    | "currency"
    | "signature"
    | "array"
    | "object"
    | "address";

type DocumentSignatureType = "signed" | "unsigned";

export type SpanPositions = {
    offset: number;
    length: number;
    iconName: string;
    pageNumber?: number;
    tableIndex?: number;
};

export type AnalyzeContentParseResult = {
    documentName: string;
    content: string;
    displayedContent: string;
    positions: Array<{
        column: number;
        lineNumber: number;
        iconName: string;
        shouldDisplayAsIcon: boolean;
        pageNumber?: number;
        tableIndex?: number;
    }>;
};

export type ParsedContentTextBlock = {
    kind?: string | undefined;
    role?: string | undefined;
    content: string;
    boundingRegions?: BoundingRegion[];
};

export type ParsedContentPagedText = {
    [pageNumber: string]: ParsedContentTextBlock[];
};

export type ParsedContentPagedTables = {
    [pageNumber: string]: StudioDocumentTable[];
};

export type ParsedContentPagedSelectionMarks = {
    [pageNumber: string]: StudioSelectionMark[];
};

export type StudioGeneralKeyValue = StudioDocumentKeyValuePair & {
    fieldColor?: string;
};

export type StudioGeneralEntityValue = {
    category: string;
    fieldColor: string;
    valueArray: StudioDocumentEntity[];
};
