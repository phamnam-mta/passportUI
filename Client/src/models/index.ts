export type SplitPaneSizes = {
    analyzeSplitPaneSize: number[];
    labelSplitPaneSize: number[];
    labelTableSplitPaneSize: number[];
};
