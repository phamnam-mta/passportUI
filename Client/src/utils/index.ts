export const delay = (ms: number) => {
    return new Promise<void>((resolve) => {
        setTimeout(() => {
            resolve();
        }, ms);
    });
};

export const getPixelWidthFromPercent = (widthInPercentage: number): number => {
    const totalWindowWidth = document.documentElement.clientWidth;
    const paneWidthInPixel = (totalWindowWidth / 100) * widthInPercentage;
    return paneWidthInPixel;
};

export const loadCanvasToBlob = (canvas: HTMLCanvasElement): Promise<Blob> => {
    return new Promise((resolve, reject) => {
        canvas.toBlob((blob) => {
            if (blob) {
                resolve(blob);
            } else {
                reject("Failed to create blob for canvas.");
            }
        });
    });
};

export const loadUrlToArrayBuffer = async (url: string): Promise<ArrayBuffer> => {
    return await fetch(url).then((r) => r.arrayBuffer());
};

export const debounce = (func, timeout = 300) => {
    return (...args) => {
        return setTimeout(() => {
            func.apply(this, args);
        }, timeout);
    };
};
