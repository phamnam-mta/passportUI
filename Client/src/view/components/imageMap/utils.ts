import { Feature } from "ol";
import { Extent } from "ol/extent";
import Polygon from "ol/geom/Polygon";
import ImageLayer from "ol/layer/Image";
import Layer from "ol/layer/Layer";
import VectorLayer from "ol/layer/Vector";
import Projection from "ol/proj/Projection";
import ImageSource from "ol/source/ImageStatic";
import VectorSource from "ol/source/Vector";
import { LAYER_NAME } from "./constants";
import { IRegion } from "./contracts";

export type Dimension = {
    width: number;
    height: number;
};

export const degreeToRadians = (degree: number) => {
    // convert degree to radians
    return (degree * Math.PI * 2) / 360;
};

export const makeLayerFilter = (layerName) => {
    return {
        layerFilter: (layer: Layer) => layer.get(LAYER_NAME) === layerName,
    };
};

export const makeVectorLayer = (layerName, options = {}): Layer => {
    return new VectorLayer({
        [LAYER_NAME]: layerName,
        source: new VectorSource(),
        ...options,
    } as any);
};

export const makeImageLayer = (layerName, uri, projection, imageExtent, options = {}): Layer => {
    return new ImageLayer({
        [LAYER_NAME]: layerName,
        source: makeImageSource(uri, projection, imageExtent),
        ...options,
    } as any);
};

export const makeImageSource = (url: string, projection: Projection, imageExtent: Extent) => {
    return new ImageSource({ url, projection, imageExtent });
};

export const convertRegionToFeature = (region: IRegion, imageExtent: Extent, isOcrProposal: boolean = false) => {
    const coordinates: Array<[number, number]> = [];
    const boundingBox = region.id.split(",").map(parseFloat);
    const imageWidth = imageExtent[2] - imageExtent[0];
    const imageHeight = imageExtent[3] - imageExtent[1];
    for (let i = 0; i < boundingBox.length; i += 2) {
        coordinates.push([Math.round(boundingBox[i] * imageWidth), Math.round((1 - boundingBox[i + 1]) * imageHeight)]);
    }
    const feature = new Feature({
        geometry: new Polygon([coordinates]),
    });
    feature.setProperties({
        id: region.id,
        text: region.value,
        highlighted: false,
        isOcrProposal,
    });
    feature.setId(region.id);

    return feature;
};

export const createRegionIdFromPolygon = (polygon: number[], page: number): string => {
    return polygon.join(",") + ":" + page;
};

export const convertToImageMapCoordinates = (polygon: number[], canvasSize: Dimension, documentSize: Dimension) => {
    const { width: canvasWidth, height: canvasHeight } = canvasSize;
    const { width: ocrWidth, height: ocrHeight } = documentSize;

    const coordinates: number[][] = [];
    for (let i = 0; i < polygon.length; i += 2) {
        coordinates.push([(polygon[i] / ocrWidth) * canvasWidth, (1 - polygon[i + 1] / ocrHeight) * canvasHeight]);
    }
    return coordinates;
};

export const getBoundingBoxFromFeatureId = (id) => id.split(",").map(parseFloat);
