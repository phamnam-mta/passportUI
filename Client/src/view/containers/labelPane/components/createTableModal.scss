.create-table-modal-body {
    padding-right: 50px;
    height: 450px;

    .table-description-container {
        margin-top: 40px;
    }
}
