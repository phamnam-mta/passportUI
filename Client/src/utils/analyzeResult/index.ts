import { BoundingRegion, Polygon, StudioFieldValue, StudioSpan } from "models/analyzeResult";

export const getPagePolygons = (boundingRegions: BoundingRegion[], page: number): Polygon[] => {
    return boundingRegions
        .filter((boundingRegion) => boundingRegion.pageNumber === page)
        .map((boundingRegion) => boundingRegion.polygon);
};

export const isAnyBoundingRegionInPage = (boundingRegions: BoundingRegion[], page: number): boolean => {
    return boundingRegions.some((boundingRegion) => boundingRegion.pageNumber === page);
};

export const getFieldTypeValue = (field: StudioFieldValue) => {
    const { type } = field;
    const typeString = type.charAt(0).toUpperCase() + type.slice(1);

    // TODO: Update when going public.
    if (type === "currency" && field.valueCurrency && typeof field.valueCurrency !== "string") {
        const { currencySymbol, currency, amount } = field.valueCurrency;
        if (currencySymbol && amount) {
            return `${currencySymbol}${amount}`;
        } else if (currency && amount) {
            return `${currency}${amount}`;
        } else if (amount) {
            return `${amount}`;
        } else {
            return undefined;
        }
    } else if (type === "address" && field.valueAddress) {
        return field.content;
    }
    return field[`value${typeString}`];
};

export interface SpanPosition extends StudioSpan {
    iconName: string;
    pageNumber?: number;
    tableIndex?: number;
    shouldDisplayAsIcon: boolean;
}
