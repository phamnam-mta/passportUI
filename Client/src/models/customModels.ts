import { FeatureCategory } from "view/components/imageMap/contracts";
import { Polygon } from "./analyzeResult";

export type CustomModel = {
    modelId: string;
    description: string;
    createdDateTime: string;
    apiVersion?: string;
    tags?: {
        modelType?: string;
        [key: string]: string | undefined;
    };
};

export type PrimitiveField = {
    fieldKey: string;
    fieldType: FieldType;
    fieldFormat: FieldFormat;
};

export type ObjectField = {
    fieldKey: string;
    fieldType: FieldType;
    fields: Field[];
    fieldFormat: FieldFormat;
    visualizationHint?: VisualizationHint;
};

export type ArrayField = {
    fieldKey: string;
    fieldType: FieldType;
    itemType: string;
    visualizationHint?: VisualizationHint;
};

export type Field = PrimitiveField | ObjectField | ArrayField;

export type FieldsWithOrder = Field & { order: number };

export enum FieldType {
    String = "string",
    Number = "number",
    Date = "date",
    Time = "time",
    Integer = "integer",
    SelectionMark = "selectionMark",
    Signature = "signature",
    Array = "array",
    Object = "object",
}

export enum FieldFormat {
    NotSpecified = "not-specified",
    Currency = "currency",
    Decimal = "decimal",
    DecimalCommaSeparated = "decimal-comma-separated",
    NoWhiteSpaces = "no-white-spaces",
    Alphanumeric = "alphanumeric",
    DMY = "dmy",
    MDY = "mdy",
    YMD = "ymd",
}

export enum VisualizationHint {
    Horizontal = "horizontal",
    Vertical = "vertical",
}

export type Labels = {
    [documentName: string]: Label[];
};

export type Label = {
    label: string;
    value: LabelValue[];
    labelType?: LabelType;
};

export type LabelValue = {
    boundingBoxes: Polygon[];
    page: number;
    text: string;
};

export type LabelValueCandidate = {
    boundingBoxes: Polygon[];
    page: number;
    text: string;
    category: FeatureCategory;
    alreadyAssignedLabelName?: string;
};

export enum LabelType {
    Words = "words",
    Region = "region",
}

export type Definitions = {
    [objectName: string]: ObjectField;
};

export enum TableType {
    dynamic = "dynamic",
    fixed = "fixed",
}

export enum HeaderType {
    row = "row",
    column = "column",
}
