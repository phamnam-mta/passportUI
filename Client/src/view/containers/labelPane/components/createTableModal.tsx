import * as React from "react";
import { Stack, TextField, Text, Image, IChoiceGroupOption, ImageFit } from "@fluentui/react";

import { constants } from "consts/constants";
import { TableType, HeaderType } from "models/customModels";
import MessageModal from "view/components/messageModal/messageModal";
import { HorizontalChoiceGroup } from "view/components/horizontalChoiceGroup/horizontalChoiceGroup";

import "./createTableModal.scss";

export interface ICreateTableModalProps {
    isOpen: boolean;
    onCreateField: (fieldKey, tableType, headerType) => Promise<void>;
    onClose: () => void;
    onGetNameErrorMessage: (value) => string | undefined;
}

export interface ICreateTableModalState {
    name: string;
    tableType: TableType | "";
    headerType: HeaderType | "";
}

export class CreateTableModal extends React.PureComponent<ICreateTableModalProps, ICreateTableModalState> {
    constructor(props) {
        super(props);
        this.state = {
            name: "",
            tableType: TableType.dynamic,
            headerType: "",
        };
    }

    private typeOptions: IChoiceGroupOption[] = [
        { key: TableType.dynamic, text: "Dynamic" },
        { key: TableType.fixed, text: "Fixed" },
    ];

    private headerOptions: IChoiceGroupOption[] = [
        { key: HeaderType.column, text: "Column" },
        { key: HeaderType.row, text: "Row" },
    ];

    private resetStates = () => {
        this.setState({ name: "", tableType: TableType.dynamic, headerType: "" });
    };

    private handleNameChange = (event: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>, newValue?: string) => {
        this.setState({ name: newValue || "" });
    };

    private handleTypeChange = (event, option) => {
        this.setState({
            tableType: option.key,
            headerType: option.key === TableType.dynamic ? "" : HeaderType.column,
        });
    };

    private handleHeaderTypeChange = (event, option) => {
        this.setState({ headerType: option.key });
    };

    private handleCloseModal = () => {
        this.resetStates();
        this.props.onClose();
    };

    private handleCreateField = async () => {
        const { onCreateField } = this.props;
        const { name, tableType, headerType } = this.state;
        await onCreateField(name, tableType, headerType);
        this.handleCloseModal();
    };

    private isCreateValid = () => {
        const { name, tableType, headerType } = this.state;
        const nameHasError = this.props.onGetNameErrorMessage(name) !== undefined;
        const isHeaderTypeValid = tableType === TableType.dynamic || !!headerType;

        return !!name && !nameHasError && !!tableType && isHeaderTypeValid;
    };

    public render() {
        const { isOpen, onGetNameErrorMessage } = this.props;
        const { name, tableType, headerType } = this.state;
        const tableTypeDescription =
            tableType === TableType.dynamic
                ? "Use dynamic tables to extract variable count of values (rows) for a given set of fields (columns)."
                : "Use fixed tables to extract specific collection of values (rows) for a given set of fields (columns and/or rows).";
        const tableTypeImgSrc =
            tableType === TableType.dynamic ? constants.dynamicTableImgSrc : constants.fixedTableImgSrc;

        return (
            <MessageModal
                isOpen={isOpen}
                onClose={this.handleCloseModal}
                title="Create table field"
                body={
                    <Stack className="create-table-modal-body" grow={1} tokens={{ childrenGap: 12 }}>
                        <TextField
                            id="name-textfield"
                            value={name}
                            onChange={this.handleNameChange}
                            onGetErrorMessage={onGetNameErrorMessage}
                            label="Name"
                            placeholder="Name your table"
                            autoComplete="off"
                            required
                        />
                        <Stack tokens={{ childrenGap: 4 }}>
                            <HorizontalChoiceGroup
                                label="Table type"
                                options={this.typeOptions}
                                selectedKey={tableType}
                                onChange={this.handleTypeChange}
                                required
                            />
                            <HorizontalChoiceGroup
                                label="Header type"
                                options={this.headerOptions}
                                selectedKey={headerType}
                                disabled={tableType === TableType.dynamic}
                                onChange={this.handleHeaderTypeChange}
                                required
                            />
                        </Stack>
                        <Stack
                            className="table-description-container"
                            horizontalAlign="center"
                            tokens={{ childrenGap: 8 }}
                        >
                            <Text>{tableTypeDescription}</Text>
                            <Image src={tableTypeImgSrc} width={650} height={200} imageFit={ImageFit.centerCover} />
                        </Stack>
                    </Stack>
                }
                width={780}
                actionButtonDisabled={!this.isCreateValid()}
                actionButtonText="Create"
                onActionButtonClick={this.handleCreateField}
            />
        );
    }
}

export default CreateTableModal;
