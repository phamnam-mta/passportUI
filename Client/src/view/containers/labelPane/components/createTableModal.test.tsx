import { TableType } from "models/customModels";
import * as React from "react";

import { flushPromises } from "utils/test";
import { HeaderType } from "models/customModels";
import { CreateTableModal } from "./createTableModal";
import { shallow } from "enzyme";

describe("<CreateTableModal />", () => {
    let baseProps;
    const mockName = "TableName";
    beforeEach(() => {
        baseProps = {
            isOpen: true,
            onClose: jest.fn(),
            onCreateField: jest.fn().mockResolvedValue({}),
            onGetNameErrorMessage: jest.fn(),
        };
    });

    describe("Rendering", () => {
        it("renders correctly", () => {
            const wrapper = shallow(<CreateTableModal {...baseProps} />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("Event Handling", () => {
        it("should reset states and close modal, when onClose is triggered", () => {
            const wrapper = shallow(<CreateTableModal {...baseProps} />);
            wrapper.setState({ name: mockName, tableType: TableType.fixed, headerType: HeaderType.column });
            const modal = wrapper.find("MessageModal") as any;
            modal.prop("onClose")();

            expect(wrapper.state("name")).toBe("");
            expect(wrapper.state("tableType")).toBe(TableType.dynamic);
            expect(wrapper.state("headerType")).toBe("");
            expect(baseProps.onClose).toBeCalledTimes(1);
        });

        it("should handle name change, when text field is changed", () => {
            const wrapper = shallow(<CreateTableModal {...baseProps} />);
            const modal = wrapper.find("MessageModal") as any;
            const textField = shallow(modal.prop("body")).find("#name-textfield");
            textField.simulate("change", {}, mockName);

            expect(wrapper.state("name")).toBe(mockName);
        });

        it("should handle name change, when text field is changed without value", () => {
            const wrapper = shallow(<CreateTableModal {...baseProps} />);
            const modal = wrapper.find("MessageModal") as any;
            const textField = shallow(modal.prop("body")).find("#name-textfield");
            textField.simulate("change", {}, undefined);

            expect(wrapper.state("name")).toBe("");
        });

        it("should handle table type change, when table type choice group is changed to fixed", () => {
            const wrapper = shallow(<CreateTableModal {...baseProps} />);
            const modal = wrapper.find("MessageModal") as any;
            const choiceGroup = shallow(modal.prop("body")).find("HorizontalChoiceGroup").first();
            choiceGroup.simulate("change", {}, { key: TableType.fixed });

            expect(wrapper.state("tableType")).toBe(TableType.fixed);
            expect(wrapper.state("headerType")).toBe(HeaderType.column);
        });

        it("should handle table type change, when table type choice group is changed to dynamic", () => {
            const wrapper = shallow(<CreateTableModal {...baseProps} />);
            wrapper.setState({ tableType: TableType.fixed, headerType: HeaderType.column });
            const modal = wrapper.find("MessageModal") as any;
            const choiceGroup = shallow(modal.prop("body")).find("HorizontalChoiceGroup").first();
            choiceGroup.simulate("change", {}, { key: TableType.dynamic });

            expect(wrapper.state("tableType")).toBe(TableType.dynamic);
            expect(wrapper.state("headerType")).toBe("");
        });

        it("should handle header type change, when header type choice group is changed to dynamic", () => {
            const wrapper = shallow(<CreateTableModal {...baseProps} />);
            wrapper.setState({ tableType: TableType.fixed, headerType: HeaderType.column });
            const modal = wrapper.find("MessageModal") as any;
            const choiceGroup = shallow(modal.prop("body")).find("HorizontalChoiceGroup").last();
            choiceGroup.simulate("change", {}, { key: HeaderType.row });

            expect(wrapper.state("headerType")).toBe(HeaderType.row);
        });

        it("should handle create field, when create button is clicked", async () => {
            const wrapper = shallow(<CreateTableModal {...baseProps} />);
            wrapper.setState({ name: mockName, tableType: TableType.fixed, headerType: HeaderType.column });
            const modal = wrapper.find("MessageModal") as any;
            modal.prop("onActionButtonClick")();

            expect(baseProps.onCreateField).toBeCalledTimes(1);
            expect(baseProps.onCreateField).toBeCalledWith(mockName, TableType.fixed, HeaderType.column);
            await flushPromises();
            expect(wrapper.state("name")).toBe("");
            expect(wrapper.state("tableType")).toBe(TableType.dynamic);
            expect(wrapper.state("headerType")).toBe("");
            expect(baseProps.onClose).toBeCalledTimes(1);
        });
    });
});
