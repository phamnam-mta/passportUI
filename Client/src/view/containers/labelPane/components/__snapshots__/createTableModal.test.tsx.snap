// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateTableModal /> Rendering renders correctly 1`] = `
<MessageModal
  actionButtonDisabled={true}
  actionButtonText="Create"
  body={
    <Stack
      className="create-table-modal-body"
      grow={1}
      tokens={
        Object {
          "childrenGap": 12,
        }
      }
    >
      <StyledTextFieldBase
        autoComplete="off"
        id="name-textfield"
        label="Name"
        onChange={[Function]}
        onGetErrorMessage={
          [MockFunction] {
            "calls": Array [
              Array [
                "",
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
        placeholder="Name your table"
        required={true}
        value=""
      />
      <Stack
        tokens={
          Object {
            "childrenGap": 4,
          }
        }
      >
        <HorizontalChoiceGroup
          label="Table type"
          onChange={[Function]}
          options={
            Array [
              Object {
                "key": "dynamic",
                "text": "Dynamic",
              },
              Object {
                "key": "fixed",
                "text": "Fixed",
              },
            ]
          }
          required={true}
          selectedKey="dynamic"
        />
        <HorizontalChoiceGroup
          disabled={true}
          label="Header type"
          onChange={[Function]}
          options={
            Array [
              Object {
                "key": "column",
                "text": "Column",
              },
              Object {
                "key": "row",
                "text": "Row",
              },
            ]
          }
          required={true}
          selectedKey=""
        />
      </Stack>
      <Stack
        className="table-description-container"
        horizontalAlign="center"
        tokens={
          Object {
            "childrenGap": 8,
          }
        }
      >
        <Text>
          Use dynamic tables to extract variable count of values (rows) for a given set of fields (columns).
        </Text>
        <Memo(Image)
          height={200}
          imageFit={4}
          src="/images/customModels/dynamic-table.png"
          width={650}
        />
      </Stack>
    </Stack>
  }
  isOpen={true}
  onActionButtonClick={[Function]}
  onClose={[Function]}
  title="Create table field"
  width={780}
/>
`;
